File: packages/gbnf/javascript/src/utils/errors/grammar-parse-error.ts
export const GRAMMAR_PARSER_ERROR_HEADER_MESSAGE = (reason: string) => `Failed to parse grammar: ${reason}`;
import {
  buildErrorPosition,
} from "./build-error-position.js";

export class GrammarParseError extends Error {
  grammar: string;
  pos: number;
  reason: string;
  constructor(grammar: string, pos: number, reason: string) {
    super([
      GRAMMAR_PARSER_ERROR_HEADER_MESSAGE(reason),
      '',
      ...buildErrorPosition(grammar, pos),
    ].join('\n'));
    this.name = 'GrammarParseError';
    this.grammar = grammar;
    this.reason = reason;
    this.pos = pos;
  }
}

File: packages/gbnf/python/gbnf/utils/errors/grammar_parse_error.py
from .build_error_position import build_error_position


def GRAMMAR_PARSER_ERROR_HEADER_MESSAGE(reason: str) -> str:
    return f"Failed to parse grammar: {reason}"


class GrammarParseError(Exception):
    grammar: str
    pos: int
    reason: str

    def __init__(self, grammar: str, pos: int, reason: str):
        super().__init__(
            "\n".join(
                [
                    GRAMMAR_PARSER_ERROR_HEADER_MESSAGE(reason),
                    "",
                    *build_error_position(grammar, pos),
                ],
            ),
        )
        self.grammar = grammar
        self.pos = pos
        self.reason = reason